odata-protocol/11.4 Data Modification.md

```diff
@@ -1040,6 +1040,10 @@ through the `ContentIDSupported` property of the
 [`Capabilities.DeepUpdateSupport`](https://github.com/oasis-tcs/odata-vocabularies/blob/main/vocabularies/Org.OData.Capabilities.V1.md#DeepUpdateSupportType)
 term, both defined in [OData-VocCap](#ODataVocCap).
 
+Clients MAY specify [ETag](#UseofETagsforAvoidingUpdateConflicts) values
+obtained from previous requests when updating entities within a collection. If an ETag 
+is provided that does not match the ETag value of the entity being updated, or if the entity does not already exist, services that support ETags MUST NOT apply the change and SHOULD indicate a response code of `412 Precondition Failed`. The special value `\*` can be used to match any existing entity but fail if the entity does not already exist.
```

Comment:
```suggestion
is provided that does not match the ETag value of the entity being updated, or if the entity does not already exist, services that support ETags MUST NOT apply the change and SHOULD indicate a response code of [`412 Precondition Failed`](#ResponseCode412PreconditionFailed). The special value `*` can be used to match any existing entity but fail if the entity does not already exist.
```
Or does the special value really contain a backslash?

Reply:
No, the backslash was an artifact when migrating from Jira issue [ODATA-1549](https://issues.oasis-open.org/browse/ODATA-1549) which does not have this backslash. Fixed the Github issue description.

Reply:
This text only describes the `continue-on-error=false` case, without calling that out.

It also doesn't state that the ETags are per entity, using the `@odata.etag` control information.

Reply:
Good catch on the backslash -- I actually thought it *was* part of the string due to some requirements on ETag format, but happy to see it is not intended.

I struggled a lot with the continue-on-error stuff. We don't introduce continue-on-error until later in the section, so my thought was that we would provide the general context (that the error is 412 Precondition Failed, without saying *how* it was returned, or whether it stops processing) and then, when the reader reads the section on continue-on-error they would just apply that knowledge to understand how the error case was handled for continue-on-error true or false.

Reply:
Suggest to add an item to the bulleted list that follows:
* Unmatched ETags within the request MUST be represented in the response as as either entities or entity references, annotated with the term `Core.DataModificationException` with a `responseCode` value of 412.

And augment the last sentence:

If the `continue-on-error` preference has not been specified, and the service is unable to apply all of the changes in the request, then it MUST return an error response _with status code 412, if appropriate,_ and MUST NOT apply any of the changes specified in the request payload.

Reply:
That would nicely capture it!

Reply:
See if the current wording helps -- I broke out the section on error handling and reference it from the etag section from proper error handling.